postcommit.py

```diff
@@ -0,0 +1,642 @@
+            end_point = 100
+        else:
+            end_point = len(self.repo)
+        for rev in xrange(self.repo[self.node].rev() + 1, end_point, 1):
+
+            c = self.repo[rev]
+            changeset = c.hex()[0:12]
+            commit_date = util.datestr(c.date(), '%Y-%m-%d %H:%M:%S')
+            parents = c.parents()
+            branch = c.branch()
+            changed_files = c.files()
+            committed_user = c.user()
+            description = c.description()
+            newchangesets.append(changeset)
+
+            commit = {'changeset' : changeset, 'branch' : branch, 'commit_date' : commit_date,
+                      'message' : description, 'user' : committed_user, 'parents' : [],
+                      'push_time': getCurrentTime()
+                      }
+
+	    #  diff_changes temporarly not going to be used *may* be useful in future (just add **self.getDiffContent(parent,node)** in diff_changes )
+            for prchang in parents:
+                parent_changeset = prchang.hex()[0:12]
+                parent_branch = self.repo[parent_changeset].branch()
+                commit['parents'].append({'changeset' :parent_changeset, 'branch' : parent_branch,
+                                          'diff_changes' : ""})
+
+            	if changed_files:
+                	committed_files = list(self.status(parent_changeset, changeset))
+                	committed_files[0] = list(set(committed_files[0]).intersection(set(changed_files)))
+                	committed_files[1] = list(set(committed_files[1]).intersection(set(changed_files)))
+                	committed_files[2] = list(set(committed_files[2]).intersection(set(changed_files)))
+            	else:
+                	committed_files = ([], [], [])
+
+            	commit['files'] = {'Added': self.diffstat(parent_changeset,changeset,committed_files[1]), 'Modified': self.diffstat(parent_changeset,changeset,committed_files[0]), 'Removed': self.diffstat(parent_changeset,changeset,committed_files[2])}
+
+            if len(commit['parents']) == 2:
+                if commit['branch'] == commit['parents'][1]['branch']:
+                    commit['commit_type'] = 3
+                else:
+                    commit['commit_type'] = 2
+            else:
+                commit['commit_type'] = 1
+
+            commits.append(commit)
+        commits.reverse()
+        return commits, newchangesets
+
+    def status(self, changeset1, changeset2):
+        return self.repo.status(changeset1, changeset2, match=None)
+
+    def getAddedFiles(self, branch):
+
+        return self.repo.status(self.branch_details[branch]['parent'], branch, match=None)[1]
+
+    def getModifedFiles(self, branch):
+
+        return self.repo.status(self.branch_details[branch]['parent'], branch, match=None)[0]
+
+    def getRemovedFiles(self, branch):
+
+        return self.repo.status(self.branch_details[branch]['parent'], branch, match=None)[2]
+
+    def getAllFiles(self, changeset):
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename method "getAllFiles" to match the regular expression ^[a-z_][a-z0-9_]{2,}$. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=python%3AS100)